### ISSUE TITLE
Multifit parameter model bug introduced in kafe2 2.3.1

### ISSUE THREAD

**Issue by pb-programmer** [dsavoiu/kafe2#171]
I noticed a bug in an analysis script after updating kafe2 to version 2.3.1 (from 2.3.0)

**Describe the bug**
Setting up a MultiFit, an AttributeError inside the kafe2 code is raised while calling MultiFit.set_parameter_values().
This behaviour was introduced in 2.3.1.

**To Reproduce**
With the latest stable kafe2 version (2.3.1), the toy example attached to the end of this issue will throw an AttributeError:

```
Script: kafe2_bug.py
Python:  3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
Kafe2:  2.3.1 
---------------------
Traceback (most recent call last):
  File "[...]\kafe2_bug.py", line 36, in <module>
    multi_fit.set_parameter_values(**dict(beta=5, n_samples=N_samples, f=0.1, a=limits[0], b=limits[1]))
  File "[...]\Python39\lib\site-packages\kafe2\fit\_base\fit.py", line 699, in set_parameter_values
    self._param_model.parameters = self.parameter_values
AttributeError: 'NoneType' object has no attribute 'parameters'

Process finished with exit code 1
```

**Expected behavior**
Downgrading to kafe2 2.3.0 "fixes" the problem
```
Script: kafe2_bug.py
Python:  3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
Kafe2:  2.3.0 
---------------------
Warning: the cost function has been evaluated as infinite. The fit might not converge correctly.
beta : 4.758e+00 +- 1.828e-01
n_samples : 9.516e-01 +- 9.301e-03
f : 1.293e-01 +- 2.452e-02
a : 0.000e+00 +- 0.000e+00
b : 1.500e+01 +- 0.000e+00

Process finished with exit code 0
```

**System:**
 - OS: Windows 10 (same problem on WSL2 Ubuntu) 
 - Python Version: 3.9.6

## Script to reproduce the bug (kafe2_bug.py)
```
import os
import sys
import matplotlib.pyplot as plt
import numpy as np
from kafe2 import HistContainer, HistFit, MultiFit, Plot
import kafe2


def exp_decay_with_background(x, beta=5, n_samples=5000, f=0.1, a=0, b=10):
    return n_samples * (1 - f) * np.exp(-x / beta) / (beta * (np.exp(-a / beta) - np.exp(-b / beta))) + n_samples * f/(b - a)


if __name__ == '__main__':
    print("Script: " + os.path.basename(sys.argv[0]))
    print("Python: ", sys.version)
    print("Kafe2: ", kafe2.__version__, "\n".ljust(22, '-'))

    # > Generate random data for fitting
    n_bins = 100
    N_samples = 5000
    limits = (0, 15)
    rng = np.random.default_rng()
    simulated_data_1 = np.concatenate((rng.exponential(scale=5, size=N_samples), rng.uniform(low=limits[0], high=limits[1], size=N_samples // 10)))
    simulated_data_2 = np.concatenate((rng.exponential(scale=5, size=N_samples), rng.uniform(low=limits[0], high=limits[1], size=N_samples // 10)))

    # > Prepare histograms and the multi fit
    histogram_1 = HistContainer(n_bins=n_bins, bin_range=limits, fill_data=simulated_data_1)
    histogram_2 = HistContainer(n_bins=n_bins, bin_range=limits, fill_data=simulated_data_2)
    exp_fit_1 = HistFit(histogram_1, exp_decay_with_background, cost_function='nllr')
    exp_fit_2 = HistFit(histogram_2, exp_decay_with_background, cost_function='nllr')

    shared_parameter_names_dict = {'beta': r'\beta', 'n_samples': r'n_{samples}', 'f': 'f', 'a': 'a', 'b': 'b'}
    exp_fit_1.assign_parameter_latex_names(**shared_parameter_names_dict)
    exp_fit_2.assign_parameter_latex_names(**shared_parameter_names_dict)

    multi_fit = MultiFit((exp_fit_1, exp_fit_2), minimizer='iminuit')

    # > Perform exponential decay (multi)fit on histogram.
    multi_fit.set_parameter_values(**dict(beta=5, n_samples=N_samples, f=0.1, a=limits[0], b=limits[1]))  # BUG!
    multi_fit.limit_parameter('f', 0, 1)
    multi_fit.fix_parameter('a', limits[0])
    multi_fit.fix_parameter('b', limits[1])
    multi_fit.do_fit()

    # > Show fitting result
    for name, value, error in zip(multi_fit.parameter_names, multi_fit.parameter_values,
                                  multi_fit.parameter_errors):
        print(name, ': {:0.3e} +- {:0.3e}'.format(value, error))

    # > Plot the resulting graph
    multi_fit = Plot(multi_fit, separate_figures=False)
    multi_fit.plot(asymmetric_parameter_errors=True)
    multi_fit.axes[0]['main'].set_xlabel(r'$t$ in s')
    plt.show()
```

**Comment by JohannesGaessler** [comment]
I made a new release with a fix.
Please confirm if this fixes the issue for you.

**Comment by pb-programmer** [comment]
> I made a new release with a fix.
> Please confirm if this fixes the issue for you.

Yes, PyPi version 2.4.0rc2 fixes the issue in my analysis script. Thank you very much for the quick response!

**Comment by cverstege** [comment]
I just released v2.3.2 with this fix. So everyone should be able get the fix via PyPi right now, instead of waiting for the upcoming 2.4 or using a pre-release.
